src/app/downloads/downloads.component.ts

```diff
@@ -38,11 +45,22 @@ export class DownloadsComponent implements OnInit, OnDestroy {
   isEpisodeLoading = true;
   isLoadingForTheFirstTime = true;
   errors: string[] = [];
-  static DONT_BREAK_CASTLE_LIMIT = 20;
+  chartPodcast: boolean;
+  chartedEpisodes: number[];
 
   constructor(private castle: CastleService,
               public store: Store<any>,
-              private router: Router) {}
+              private router: Router,
+              private route: ActivatedRoute) {
+    route.params.subscribe(params => {
+      if (params['chartPodcast']) {
+        this.chartPodcast = params['chartPodcast'] === 'true';
+      }
+      if (params['episodes']) {
+        this.chartedEpisodes = params['episodes'].split(',').map(id => +id);
```

Comment:
Seeing a thing where if you uncheck all episodes, and then recheck any of them, the last episode you unchecked will show as checked again. To solve I think you need an else this.chartedEpisodes = [] here.

Reply:
Good catch. This stuff is really convoluted without doing router state composition.